libs/application/templates/parental-leave/src/hooks/useGetOrRequestEndDates.ts

```diff
@@ -1,118 +1,76 @@
+      startDate: string
+      lengthInMonths: number
+    }) => {
+      const id = `${startDate}/${lengthInMonths}`
 
       if (loadedEndDates.has(id)) {
         setLoading(false)
 
-        return loadedEndDates.get(id)!
+        return loadedEndDates.get(id)
       }
 
-      const temporaryPercentage = '100'
-
-      const { data: temporaryEndDateData } = await lazyGetEndDate({
-        input: {
-          startDate,
-          length: String(length),
-          percentage: temporaryPercentage,
-        },
-      })
+      const start = parseISO(startDate)
 
-      const temporaryLazyEndDate =
-        temporaryEndDateData?.getParentalLeavesPeriodEndDate?.periodEndDate
+      const wholeMonthsToAdd = Math.floor(lengthInMonths)
+      const daysToAdd =
+        (Math.round((lengthInMonths - wholeMonthsToAdd) * 100) / 100) * 30
 
-      if (!temporaryLazyEndDate) {
-        setLoading(false)
-
-        throw new Error(
-          `VMST: Cannot calculate temporary end date, startDate/${startDate} length/${length} temporaryLazyEndDate/${temporaryLazyEndDate}`,
-        )
-      }
+      const end = addDays(addMonths(start, wholeMonthsToAdd), daysToAdd - 1)
 
-      const { data: lengthData } = await lazyGetLength({
-        input: {
-          startDate,
-          endDate: new Date(temporaryLazyEndDate).toISOString(),
-          percentage: temporaryPercentage,
-        },
-      })
-
-      const startToEndDatesLength =
-        lengthData?.getParentalLeavesPeriodLength.periodLength
-
-      if (!startToEndDatesLength) {
-        setLoading(false)
-
-        throw new Error(
-          `VMST: Cannot calculate length between startDate and calculated endDate, startDate/${startDate} length/${length} temporaryLazyEndDate/${temporaryLazyEndDate} startToEndDatesLength/${startToEndDatesLength}`,
-        )
-      }
-
-      const computedPercentage = calculateDaysToPercentage(
-        application,
-        startToEndDatesLength,
-      )
-
-      const computedLength = Math.min(
-        100,
-        Math.floor((length * computedPercentage) / 100),
+      const maxPercentage = calculateMaxPercentageForPeriod(
+        start,
+        end,
+        remainingRights,
       )
+      const length = calculatePeriodLength(start, end, maxPercentage)
 
-      const { data } = await lazyGetEndDate({
-        input: {
-          startDate,
-          length: String(computedLength),
-          percentage: String(computedPercentage),
-        },
-      })
-
-      const lazyEndDate = data?.getParentalLeavesPeriodEndDate?.periodEndDate
-
-      if (!lazyEndDate) {
-        setLoading(false)
-
-        throw new Error(
-          `VMST: Cannot calculate end date, startDate/${startDate} length/${length} temporaryLazyEndDate/${temporaryLazyEndDate} startToEndDatesLength/${startToEndDatesLength} computedPercentage/${computedPercentage} lazyEndDate/${lazyEndDate}`,
-        )
+      const result = {
+        date: end,
+        percentage: maxPercentage * 100,
```

Comment:
I would probably do the `* 100` directly within the function to return a "percentage"